daemon/container.go

```diff
@@ -305,6 +305,11 @@ func validateHostConfig(hostConfig *containertypes.HostConfig) error {
 	if !hostConfig.Isolation.IsValid() {
 		return errors.Errorf("invalid isolation '%s' on %s", hostConfig.Isolation, runtime.GOOS)
 	}
+	for k := range hostConfig.Annotations {
+		if k == "" {
+			return errors.Errorf("invalid Annotations: the empty string is not permitted as an annotation key")
```

Comment:
I had a look if we should be using an `errdefs.InvalidParameter()` here (and noticed other errors didn't), but it looks like we wrap any errors returned here on the call sites, such as https://github.com/moby/moby/blob/2f0e308c7d4cb59f33664b763fa95542dee3c024/daemon/create.go#L65-L68


So in either case; it's good (currently) to not use an errdefs, but we _could_ (should?) look at these at some points to assign error-types in these locations.